FirebaseStorageSwift/Sources/SwiftAPIExtension.swift

```diff
@@ -56,16 +101,60 @@ public extension StorageReference {
          completion: getResultCallback(completion: completion))
   }
 
+  /**
+    List up to `maxResults` items (files) and prefixes (folders) under this StorageReference.
+
+    "/" is treated as a path delimiter. Firebase Storage does not support unsupported object
+    paths that end with "/" or contain two consecutive "/"s. All invalid objects in GCS will be
+    filtered.
+
+   Only available for projects using Firebase Rules Version 2.
```

Comment:
Maybe specify what happens if you call this (and the other list function) on a project using rules v1.

Reply:
Out of scope for this PR. The same open question applies to the original list API docs.